sw/host/opentitanlib/src/app/mod.rs

```diff
@@ -93,7 +101,7 @@ impl ProgressIndicator for StagedProgressBar {
     fn progress(&self, pos: usize) {
         let bar = self.current_progress_bar.borrow();
         let bar = bar.as_ref().unwrap();
-        if pos as u64 == bar.length() {
+        if pos as u64 == bar.length().unwrap() {
```

Comment:
Are the bar.length unwraps safe? When is [`BarState.len`](https://docs.rs/indicatif/0.17.6/src/indicatif/state.rs.html#25) None?

Reply:
It's none when a spinner is created, but we create it with `ProgressBar::new` so this should always return a value.